src/test/java/com/titarenko/rest/HttpControllerTest_withoutMocks.java

```diff
@@ -12,68 +13,207 @@
+            .withDateOfHire(LocalDate.parse("2011-11-11"))
+            .build();
+    private static Employee stepa = new Employee.Builder()
+            .withId(3)
+            .withName("Stepa")
+            .withSex(Gender.M)
+            .withPosition("position3")
+            .withSalary(200)
+            .withDateOfHire(LocalDate.parse("2010-10-10"))
+            .build();
+
+    private String incomingJson = """
+                {
+                  "id" : 10,
+                  "name" : "Tom Cruz",
+                  "sex" : "M",
+                  "position" : "actor",
+                  "salary" : 350,
+                  "dateOfHire" : "2010-10-10"
+                }
+                """;
 
     @BeforeAll
     static void startHttpController() {
+        database = new ArrayList<>();
+        Collections.addAll(database, vasil, petr, stepa);
         new Controller(new EmployeeServiceImpl(new EmployeeDao() {
             @Override
             public Integer create(Employee employee) {
-                return null;
+                database.add(employee);
+                return database.get(database.size() - 1).getId();
             }
 
             @Override
             public Employee get(String name) {
-                return null;
+                Employee returned = null;
+                for (Employee empl : database) {
+                    if (name.equals(empl.getName())) {
+                        returned = empl;
+                    }
+                }
+                return returned;
             }
 
             @Override
             public Employee get(int id) {
-                return null;
+                Employee returned = null;
+                for (Employee empl : database) {
+                    if (id == empl.getId()) {
+                        returned = empl;
+                    }
+                }
+                return returned;
             }
 
             @Override
             public Employee update(Integer id, Employee employee) {
-                return null;
+                employee.setId(id);
+                int index = database.indexOf(get(id));
+                database.set(index, employee);
+                return database.get(index);
             }
 
             @Override
             public boolean delete(String name) {
-                return false;
+                int initSize = database.size();
+                database.remove(get(name));
+                return initSize == database.size() + 1;
             }
 
             @Override
             public List<Employee> getAll() {
-                return Collections.singletonList(vasil);
+                return database;
             }
 
             @Override
             public List<Employee> getEmployeesWithSameSalary() {
-                return null;
+                Map<Integer, List<Employee>> map = new HashMap<>();
+                for (Employee empl : database) {
+                    if (map.containsKey(empl.getSalary())) {
```

Comment:
There is no need for such complex logic.
`return Arrays.asList(vasil, petr);` would be enough.